firstvoices/backend/views/sites_views.py

```diff
@@ -93,6 +108,35 @@ def list(self, request, *args, **kwargs):
 
         return Response(data)
 
+    def update(self, request, *args, **kwargs):
+        request_data = request.data
+        site_slug = kwargs["slug"]
+        logo_id = request_data["logo"]
+        banner_image_id = request_data["banner_image"]
+        banner_video_id = request_data["banner_video"]
+
+        # Verifying if the media belongs to the site
+        if logo_id and not verify_media_source(site_slug, "image", logo_id):
+            raise ValidationError("Logo supplied is not from the specified site.")
```

Comment:
Validation like this belongs in the serializer instead. In this case, you should be able to reuse the existing SameSite validator on each of the media fields.